################################################################################
SHORT DESCRIPTION: 
################################################################################
Disabling interrupts early causes debug assertions.

################################################################################
LONG DESCRIPTION: 
################################################################################
This is only seen with debug builds but since it causes assertions it is
probably a bigger problem. It clearly says in map_pages_to_xen that it should
not be called with interrupts disabled. Moved disabling to just after that
call.

################################################################################
CHANGELOG 
################################################################################

################################################################################
REMOVAL 
################################################################################

################################################################################
UPSTREAM PLAN
################################################################################
This patch has been submitted to upstream Xen by Christopher Clark
and accepted in upstream commit: 80eb3da01bc4378af537d60e3d55767acf0d16ea

This patch can be removed from OpenXT in the next major version upgrade of Xen.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################

################################################################################
PATCHES 
################################################################################
--- a/xen/arch/x86/tboot.c
+++ b/xen/arch/x86/tboot.c
@@ -341,8 +341,6 @@ void tboot_shutdown(uint32_t shutdown_ty
 
     g_tboot_shared->shutdown_type = shutdown_type;
 
-    local_irq_disable();
-
     /* Create identity map for tboot shutdown code. */
     /* do before S3 integrity because mapping tboot may change xenheap */
     map_base = PFN_DOWN(g_tboot_shared->tboot_base);
@@ -357,6 +355,10 @@ void tboot_shutdown(uint32_t shutdown_ty
         return;
     }
 
+    /* Disable interrupts as early as possible but not before */
+    /* calling map_pages_to_xen */
+    local_irq_disable();
+
     /* if this is S3 then set regions to MAC */
     if ( shutdown_type == TB_SHUTDOWN_S3 )
     {
